docs/rules/restrict-dependency-ranges.md

```diff
@@ -0,0 +1,114 @@
+# restrict-dependency-ranges
+
+💡 This rule is manually fixable by [editor suggestions](https://eslint.org/docs/latest/use/core-concepts#rule-suggestions).
+
+<!-- end auto-generated rule header -->
+
+This rule allows you to require that specific dependencies use a particular kind
+of semver range (e.g. `^`). There are several options for specifying which dependencies
+a range type restriction should be applied to, including dependency type,
+package name (or name regex pattern), and version range (e.g. '<1').
+
+If you provide multiple options and a dependency matches more than one of the
+options, the last option that matches will take precedent for that dependency.
+This allows you to define more general rules that apply to all dependencies (or large
+groups of dependencies), and then define more granular options to apply exceptions
+or focus on some subset of dependencies.
+
+## Options
+
+`type DependencyType = 'dependencies' | 'devDependencies' | 'optionalDependencies' | 'peerDependencies';`\
+`type RangeType = 'caret' | 'pin' | 'tilde';`
+
+| Name                 | Type                     | Required |
+| :------------------- | :----------------------- | :------- |
+| `forDependencyTypes` | DependencyType[]         |          |
+| `forPackages`        | String[]                 |          |
+| `forVersions`        | String                   |          |
```

Comment:
[Docs] To be more correct with [`@typescript-eslint/no-wrapper-object-types`](https://typescript-eslint.io/rules/no-wrapper-object-types):

```suggestion
| `forPackages`        | string[]                 |          |
| `forVersions`        | string                   |          |
```

Reply:
Updated